Realm/RLMUtil.mm

```diff
@@ -240,6 +241,44 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
     return outArray;
 };
 
+NSArray *RLMCollectionValueForKey(NSString *key, RLMRealm *realm, RLMObjectSchema *objectSchema, size_t count, size_t (^indexGenerator)()) {
+    if (count == 0) {
+        return @[];
+    }
+    BOOL keyIsSelf = [key isEqualToString:@"self"];
+    NSMutableArray *results = [NSMutableArray arrayWithCapacity:count];
+    RLMObjectBase *accessor = RLMCreateObjectAccessor(realm, objectSchema, 0);
```

Comment:
We need to call RLMCreateObjectAccessor for every element for swift list properties to work. Once we do this we no longer have to get the first row twice.